crates/api_common/src/request.rs

```diff
@@ -63,47 +64,40 @@ pub async fn fetch_link_metadata(url: &Url, context: &LemmyContext) -> LemmyResu
     .await?
     .error_for_status()?;
 
-  // In some cases servers send a wrong mime type for images, which prevents thumbnail
-  // generation. To avoid this we also try to guess the mime type from file extension.
-  let content_type = mime_guess::from_path(url.path())
-    .first()
-    // If you can guess that its an image type, then return that first.
-    .filter(|guess| guess.type_() == mime::IMAGE)
-    // Otherwise, get the content type from the headers
-    .or(
-      response
-        .headers()
-        .get(CONTENT_TYPE)
-        .and_then(|h| h.to_str().ok())
-        .and_then(|h| h.parse().ok()),
-    );
-
-  let opengraph_data = {
-    // if the content type is not text/html, we don't need to parse it
-    let is_html = content_type
-      .as_ref()
-      .map(|c| {
-        (c.type_() == mime::TEXT && c.subtype() == mime::HTML)
-      ||
-      // application/xhtml+xml is a subset of HTML
-      (c.type_() == mime::APPLICATION && c.subtype() == "xhtml")
-      })
-      .unwrap_or(false);
-    if !is_html {
-      Default::default()
-    } else {
+  let mut content_type: Option<Mime> = response
+    .headers()
+    .get(CONTENT_TYPE)
+    .and_then(|h| h.to_str().ok())
+    .and_then(|h| h.parse().ok());
+
+  let mut opengraph_data = Default::default();
+
+  if let Some(c) = &content_type {
+    if (c.type_() == mime::TEXT && c.subtype() == mime::HTML)
+	    ||
+	    // application/xhtml+xml is a subset of HTML
+	    (c.type_() == mime::APPLICATION && c.subtype() == "xhtml")
```

Comment:
No predefined constant here, but you define a variable `TEXT_XHTML` and use that for comparison to make it shorter. Also move the comment before the if, not in the middle.

Reply:
"+xml" needs to be included in the constant, and comparison for both html and xhtml should only use the essence string (`c.essense()`) to ignore parametters

Reply:
This can't be a const unfortunately due to rust-lang/rust#74935

Reply:
If we compare only the essence string then there is no need to include `+xml` in the constant as it gets ignored, right? Anyway just a minor nitpick.